scripts/control_bar.js

```diff
@@ -322,28 +316,64 @@ function makeImageWidget(container, align = "C") {
     let cosine = 1;
 
     const percentInput = $("<input>", {type: 'number', value: percent, title: texts.zoomPercent});
+    let contentWidth, imageWidth, imDivWidth, vertOffset;
 
     function setImageStyle() {
-        // when this is called in 'setImage' we will not know the dimensions
-        // of the image but it is not rotated (sine=0) so we do not need to.
-        let offset;
+        // To allow scrolling when image does not overflow the window
+        // make imageDiv bigger than image, different options are:
+        // windowWidth: no extra scrolling,
+        // (2 * windowWidth - imageWidth): scroll to edges,
+        // (2 * windowWidth): half left to half right,
+        // (2 * windowWidth + imageWidth): scroll to far edges,
+        // similarly for vertical
+        contentWidth = parseFloat(getComputedStyle(content).width);
+        const contentHeight = parseFloat(getComputedStyle(content).height);
         image.style.width = `${10 * percent}px`;
         image.style.height = "auto";
+        let imageHeight, xOffset, yOffset, imDivHeight;
         if(sine != 0) {
             // rotated 90 or 270 degrees
-            imageDiv.style.height = `${image.width}px`;
-            imageDiv.style.width = `${image.height}px`;
-            offset = (image.height - image.width) / 2;
+            imageWidth = image.height;
+            imageHeight = image.width;
+            yOffset = - (image.height - image.width) / 2;
         } else {
             // rotated 0 or 180 degrees
-            imageDiv.style.height = "auto";
-            imageDiv.style.width = "auto";
-            offset = 0;
+            imageWidth = image.width;
+            imageHeight = image.height;
+            yOffset = 0;
         }
+        // center aligned so:
+        xOffset = 0;
+        imDivWidth = Math.max(2 * (contentWidth - 60) + imageWidth, imageWidth);
+        imageDiv.style.width = `${imDivWidth}px`;
+        imDivHeight = Math.max(2 * (contentHeight - 60) + imageHeight, imageHeight);
+        imageDiv.style.height = `${imDivHeight}px`;
+        // adjust yOffset to be able to scroll downwards
+        vertOffset = Math.max(0.5 * (imDivHeight - imageHeight), 0);
+        yOffset += vertOffset;
         // image rotates about centre. Offset moves it to correct position
-        image.style.transform = `matrix(${cosine}, ${-sine}, ${sine}, ${cosine}, ${offset}, ${-offset})`;
+        image.style.transform = `matrix(${cosine}, ${-sine}, ${sine}, ${cosine}, ${xOffset}, ${yOffset})`;
+    }
+
+    function initScroll() {
+        if(align == "C") {
```

Comment:
since `"C"` is used twice, should we keep a constant for it at least? Even if we never reference any others such as left or right anymore?

Reply:
I suppose we should at least check that the value provided is in the set we have allowed for.

Reply:
I don't think we need to change this. Nothing bad can happen.